iwyu.cc

```diff
@@ -4403,16 +4403,9 @@ int main(int argc, char **argv) {
   //   path/to/iwyu -Xiwyu --verbose=4 [-Xiwyu --other_iwyu_flag]... \
   //       CLANG_FLAGS... foo.cc
   OptionsParser options_parser(argc, argv);
-
-  std::unique_ptr<clang::CompilerInstance> compiler(CreateCompilerInstance(
-      options_parser.clang_argc(), options_parser.clang_argv()));
-  if (!compiler) {
+  if (!ExecuteAction(options_parser.clang_argc(), options_parser.clang_argv(),
+                     []() { return std::make_unique<IwyuAction>(); })) {
```

Comment:
Is the lambda function/factory here necessary? To me it seems easier and more obvious if we just pass `std::make_unique<IwyuAction>()` into `ExecuteAction`.

Reply:
I have a few PRs buffered where it will be necessary to delay construction:

* ... to be able to pass in ToolChain into IwyuAction constructor
* ... to sometimes avoid IwyuAction entirely in favor of a real preprocessing job

so yeah, this looks a little overengineered currently, but it will make the coming PRs easier to understand.

Reply:
Ack.